python/ray/dag/compiled_dag_node.py

```diff
@@ -1728,9 +1736,14 @@ def teardown(self, wait: bool, kill_actors: bool = False):
 
                 if not do_teardown:
                     # Teardown is already being performed.
-                    if wait:
-                        self.wait_teardown(kill_actors)
-                    return
+                    # self.wait_teardown(kill_actors)
+                    while True:
```

Comment:
this was the root cause of segfault

I think there was a dangling monitor thread that was calling this API and waiting for ray.get, which is triggering segfault when the driver is shutdown

Reply:
> calling this API and waiting for ray.get, which is triggering segfault when the driver is shutdown

1. Why this function will be called after the driver process is shutdown?
2. This function still calls `ray.get` and `self.wait_teardown()` below. Why does the issue disappear?

Reply:
1. it is not called "after" shutdown. but when shutdown happens, 2 threads (main and monitor) get into this wait_teardown, which calls ray.get. If the main thread one finishes, the shutdown finishes and the next test starts. If the monitor thread one is not finished before the main thread, the dangling thread still is waiting for ray.get, which basically crashes because ray is shutdown and core worker doesn't exist

Reply:
Thank you for the explanation!